app/models/calculator_resolver.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+class CalculatorResolver
+  def self.call(calculator)
+    dependencies = {}
+    get_fields(calculator).each do |field|
+      unless field[:type] == 'Calculation'
+        dependencies[field] = []
+        next
+      end
+      dependencies[field] = get_dependencies(field[:value])
+    end
+
+    dependencies
+  end
+
+  def self.get_dependencies(value)
+    fetched_values = []
+    calculator = Dentaku::Calculator.new
+    selectors = calculator.dependencies(value)
+    selectors.each do |selector|
+      fetched_values << Value.where(selector: selector.upcase)[0]
+    end
+
+    fetched_values
+  end
+
+  def self.get_fields(calculator)
+    Field.joins(:calculator)
```

Comment:
What's the purpose of joins here?

Reply:
As I've understood, I was looking for a solution in the wrong direction. So I've decided to return to first solution and to replace `calculator_id: calculator.id` on `calculator: calculator.id` in where clause.